WordPress/WordPressTest/PostRepositoryPostsListTests.swift

```diff
@@ -0,0 +1,191 @@
+import Foundation
+import XCTest
+import OHHTTPStubs
+
+@testable import WordPress
+
+class PostRepositoryPostsListTests: CoreDataTestCase {
+
+    var repository: PostRepository!
+    var blogID: TaggedManagedObjectID<Blog>!
+
+    override func setUp() async throws {
+        repository = PostRepository(coreDataStack: contextManager)
+
+        let loggedIn = try await signIn()
+        blogID = try await contextManager.performAndSave {
+            let blog = try BlogBuilder($0)
+                .with(dotComID: 42)
+                .withAccount(id: loggedIn)
+                .build()
+            return TaggedManagedObjectID(blog)
+        }
+    }
+
+    override func tearDown() async throws {
+        HTTPStubs.removeAllStubs()
+    }
+
+    func testPagination() async throws {
+        // Given there are 15 published posts on the site
+        try await preparePostsList(type: "post", total: 15)
+
+        // When fetching all of the posts
+        let firstPage = try await repository.paginate(type: Post.self, statuses: [.publish], offset: 0, number: 10, in: blogID)
+        let secondPage = try await repository.paginate(type: Post.self, statuses: [.publish], offset: 10, number: 10, in: blogID)
+
+        XCTAssertEqual(firstPage.count, 10)
+        XCTAssertEqual(secondPage.count, 5)
+
+        // All of the posts are saved
+        let total = await contextManager.performQuery { $0.countObjects(ofType: Post.self) }
+        XCTAssertEqual(total, 15)
+    }
+
+    func testSearching() async throws {
+        // Given there are 15 published posts on the site
+        try await preparePostsList(type: "post", total: 15)
+
+        // When fetching all of the posts
+        let _ = try await repository.paginate(type: Post.self, statuses: [.publish], offset: 0, number: 15, in: blogID)
+
+        // There should 15 posts saved locally before performing search
+        var total = await contextManager.performQuery { $0.countObjects(ofType: Post.self) }
+        XCTAssertEqual(total, 15)
+
+        // Perform search
+        let postIDs: [TaggedManagedObjectID<Post>] = try await repository.search(input: "1", statuses: [.publish], limit: 1, orderBy: .byDate, descending: true, in: blogID)
+        XCTAssertEqual(postIDs.count, 1)
+
+        // There should still be 15 posts after the search: no local posts should be deleted
+        total = await contextManager.performQuery { $0.countObjects(ofType: Post.self) }
+        XCTAssertEqual(total, 15)
+    }
+
+}
```

Comment:
As mentioned in the `preparePostsList` API doc, these two parameters are ignored intentionally. I didn't explain there why though. It's because I don't think implementing sorting would mean much to the test case. Asserting the returned posts are in expected order doesn't test any app code. The app code doesn't really care about the order: it blindly trusts the API would return posts in the correct order.

Reply:
Oh, right right. I got confused, my bad.

I feel like if the parameters are given, we should somehow test they are passed along correctly... 🤔 

Then again, as far as business logic goes, that's quite straightforward. The cost of debugging if a mistake is made should be low enough.